src/ensembl/src/shared/components/external-reference/ExternalReference.test.tsx

```diff
@@ -15,55 +15,67 @@
  */
 
 import React from 'react';
-import { mount } from 'enzyme';
+import { render, screen } from '@testing-library/react';
 import faker from 'faker';
 
 import ExternalReference, { ExternalReferenceProps } from './ExternalReference';
-import ExternalLink from '../external-link/ExternalLink';
 
 const defaultProps: ExternalReferenceProps = {
-  label: faker.random.words(),
-  linkText: faker.random.words(),
+  label: faker.random.word(),
+  linkText: faker.random.word(),
   to: faker.internet.url(),
   classNames: {
-    container: faker.random.words(),
-    label: faker.random.words(),
-    icon: faker.random.words(),
-    link: faker.random.words()
+    container: faker.random.word(),
+    label: faker.random.word(),
+    icon: faker.random.word(),
+    link: faker.random.word()
   }
 };
 
 describe('<ExternalReference />', () => {
   const renderExternalReference = (
     props: Partial<ExternalReferenceProps> = {}
-  ) => mount(<ExternalReference {...defaultProps} {...props} />);
+  ) => render(<ExternalReference {...defaultProps} {...props} />);
 
-  let wrapper: any;
+  let container: any;
 
   beforeEach(() => {
     jest.resetAllMocks();
-    wrapper = renderExternalReference();
+    container = renderExternalReference().container;
   });
 
   it('renders without error', () => {
-    expect(() => wrapper).not.toThrow();
+    expect(() => container).not.toThrow();
   });
 
   it('hides label container div when there is no label', () => {
-    wrapper = renderExternalReference({ label: undefined });
-    expect(wrapper.find('.label')).toHaveLength(0);
+    container = renderExternalReference({ label: undefined }).container;
+    expect(container.querySelectorAll('.label')).toHaveLength(0);
   });
 
   it('applies the passed in classNames', () => {
     expect(
-      wrapper.render().hasClass(defaultProps.classNames?.container)
+      screen
+        .getByTestId('external reference container')
+        .classList.contains(defaultProps.classNames?.container as string)
```

Comment:
Is the `as string` required? It isn't specified in the next `expect` (`classList.contains(defaultProps.classNames?.label) `).

Reply:
If i removed it, typescript complains with the following error:
`(property) container?: string | undefined
Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.ts(2345)`
  
  i am also confused as to why some of the line it complains others it doesn't. It doesnt complain in the next `expect`